features/steps/steps.py

```diff
@@ -19,7 +19,7 @@ def given_a_trusty_lxd_container(context):
 
 @when("I run `{command}` {user_spec}")
 def when_i_run_command(context, command, user_spec):
-    prefix = []
+    prefix = get_command_prefix_for_user_spec(user_spec)
     if user_spec == "with sudo":
```

Comment:
I think you can drop this if/else now as you are setting prefix properly on the line above